src/textual/_clock.py

```diff
@@ -0,0 +1,58 @@
+import asyncio
+from time import monotonic
+
+
+"""
+A module that serves as the single source of truth for everything time-related in a Textual app.
+Having this logic centralised makes it easier to simulate time in integration tests, 
+by mocking the few functions exposed by this module.
+"""
+
+
+# N.B. This class and its singleton instance have to be hidden APIs because we want to be able to mock time,
+# even for Python modules that imported functions such as `get_time` *before* we mocked this internal _Clock.
+# (so mocking public APIs such as `get_time` wouldn't affect direct references to then that were done during imports)
+class _Clock:
+    def get_time(self) -> float:
+        return monotonic()
+
+    async def aget_time(self) -> float:
+        return self.get_time()
+
+    async def sleep(self, seconds: float) -> None:
+        await asyncio.sleep(seconds)
+
+
+# That's our target for mocking time! :-)
+_clock = _Clock()
+
+
+def get_time() -> float:
+    """
+    Get the current wall clock time.
+
+    Returns:
+        float: the value (in fractional seconds) of a monotonic clock, i.e. a clock that cannot go backwards.
+    """
+    return _clock.get_time()
+
+
+async def aget_time() -> float:
```

Comment:
I'm not a fan at all either to be honest :sweat_smile:  - I just opted for that `a` prefix because it seems to become the idiomatic way to make things "asyncio-related" in Python?

i.e. `object.__enter__` and `object.__exit__`  become `object.__aenter__` and `object.__aexit__` for async contexts in the Python data model.

And we find the same pattern in Django now: `aget_or_create`, `acount`, `aaggregate`, etc .
> https://github.com/django/django/pull/14843

Reply:
Consider using the "await me maybe" pattern 👀

https://simonwillison.net/2020/Sep/2/await-me-maybe/

Reply:
> Consider using the "await me maybe" pattern eyes
> 
> https://simonwillison.net/2020/Sep/2/await-me-maybe/

This pattern is good, and we use it already in Textual (for event listeners for example) :slightly_smiling_face: , but the problem with it is that the caller must be in an async context itself to use it - which is not the case for some calls we need to make to these clock-related functions, as they can be called in some classes' constructors for example